modules/jolt_physics/objects/jolt_area_3d.h

```diff
@@ -0,0 +1,228 @@
+
+	bool is_monitorable() const { return monitorable; }
+	void set_monitorable(bool p_monitorable);
+
+	bool can_monitor(const JoltBody3D &p_other) const;
+	bool can_monitor(const JoltSoftBody3D &p_other) const;
+	bool can_monitor(const JoltArea3D &p_other) const;
+
+	virtual bool can_interact_with(const JoltBody3D &p_other) const override;
+	virtual bool can_interact_with(const JoltSoftBody3D &p_other) const override;
+	virtual bool can_interact_with(const JoltArea3D &p_other) const override;
+
+	virtual Vector3 get_velocity_at_position(const Vector3 &p_position) const override;
+
+	virtual bool reports_contacts() const override { return false; }
+
+	bool is_point_gravity() const { return point_gravity; }
+	void set_point_gravity(bool p_enabled);
+
+	float get_priority() const { return priority; }
+	void set_priority(float p_priority) { priority = p_priority; }
+
+	float get_gravity() const { return gravity; }
+	void set_gravity(float p_gravity);
+
+	float get_point_gravity_distance() const { return point_gravity_distance; }
+	void set_point_gravity_distance(float p_distance);
+
+	float get_linear_damp() const { return linear_damp; }
+	void set_area_linear_damp(float p_damp) { linear_damp = p_damp; }
+
+	float get_angular_damp() const { return angular_damp; }
+	void set_area_angular_damp(float p_damp) { angular_damp = p_damp; }
+
+	OverrideMode get_gravity_mode() const { return gravity_mode; }
+	void set_gravity_mode(OverrideMode p_mode);
+
+	OverrideMode get_linear_damp_mode() const { return linear_damp_mode; }
+	void set_linear_damp_mode(OverrideMode p_mode) { linear_damp_mode = p_mode; }
+
+	OverrideMode get_angular_damp_mode() const { return angular_damp_mode; }
+	void set_angular_damp_mode(OverrideMode p_mode) { angular_damp_mode = p_mode; }
+
+	Vector3 get_gravity_vector() const { return gravity_vector; }
+	void set_gravity_vector(const Vector3 &p_vector);
+
+	Vector3 compute_gravity(const Vector3 &p_position) const;
+
+	void body_shape_entered(const JPH::BodyID &p_body_id, const JPH::SubShapeID &p_other_shape_id, const JPH::SubShapeID &p_self_shape_id);
+	bool body_shape_exited(const JPH::BodyID &p_body_id, const JPH::SubShapeID &p_other_shape_id, const JPH::SubShapeID &p_self_shape_id);
+
+	void area_shape_entered(const JPH::BodyID &p_body_id, const JPH::SubShapeID &p_other_shape_id, const JPH::SubShapeID &p_self_shape_id);
+	bool area_shape_exited(const JPH::BodyID &p_body_id, const JPH::SubShapeID &p_other_shape_id, const JPH::SubShapeID &p_self_shape_id);
+
+	bool shape_exited(const JPH::BodyID &p_body_id, const JPH::SubShapeID &p_other_shape_id, const JPH::SubShapeID &p_self_shape_id);
+	void body_exited(const JPH::BodyID &p_body_id, bool p_notify = true);
+	void area_exited(const JPH::BodyID &p_body_id);
+
+	void call_queries(JPH::Body &p_jolt_body);
+
+	virtual bool has_custom_center_of_mass() const override { return false; }
+	virtual Vector3 get_center_of_mass_custom() const override { return { 0, 0, 0 }; }
```

Comment:
```suggestion
	virtual Vector3 get_center_of_mass_custom() const override { return Vector3(0, 0, 0); }
```
Be explicit about return types